back/api/views.py

```diff
@@ -88,6 +128,22 @@ def get(self, request):
         serializer = UserSerializer(user)
         return Response(serializer.data)
 
+class TeacherView(APIView):
+    def get(self, request):
+        token = request.COOKIES.get('jwt')
+        if not token:
+            raise AuthenticationFailed('Unauthentication')
+        
+        try:
+            payload = jwt.decode(token, 'secret', algorithms=['HS256'])
+
+        except jwt.ExpiredSignatureError:
+            raise AuthenticationFailed('Unauthentication')
+        
+        teacher = Teacher.objects.get(user__id=payload['id'])
+        serializer = TeacherSerializer(teacher)
+        return Response(serializer.data)
```

Comment:
I'm surprised you would need to write this boilerplate code.

Reply:
I don't understand the comment

Reply:
Indeed, the use of a decorator is particularly suitable.